src/test/resources/Bad.scala

```diff
@@ -0,0 +1,5 @@
+import shapeless._
+
+class Bad {
+  implicit val b@bar@ar = cachedImplicit[Int]
```

Comment:
My comment above was based on the understanding that this example is actually allowed. If not, then disregard the other comment :)